src/vsr/grid_scrubber.zig

```diff
@@ -0,0 +1,481 @@
+                            table_info.checksum,
+                        );
+                    }
+
+                    const tables = tour.table_index.tables;
+                    tour.* = .{ .table_data = .{
+                        .index_checksum = table_info.checksum,
+                        .index_address = table_info.address,
+                        .tables = tables,
+                    } };
+
+                    return .{
+                        .block_checksum = table_info.checksum,
+                        .block_address = table_info.address,
+                        .block_type = .index,
+                    };
+                } else {
+                    tour.* = .{ .manifest_log = .{} };
+                }
+            }
+
+            if (tour.* == .manifest_log) {
+                if (tour.manifest_log.iterator.next(
+                    &scrubber.forest.manifest_log,
+                )) |block_reference| {
+                    return .{
+                        .block_checksum = block_reference.checksum,
+                        .block_address = block_reference.address,
+                        .block_type = .manifest,
+                    };
+                } else {
+                    tour.* = .{ .free_set = .{} };
+                }
+            }
+
+            if (tour.* == .free_set) {
+                const free_set_trailer = &scrubber.forest.grid.free_set_checkpoint;
+                if (free_set_trailer.callback != .none) return null;
+                if (tour.free_set.index < free_set_trailer.block_count()) {
+                    const index = tour.free_set.index;
+                    tour.free_set.index += 1;
+                    return .{
+                        .block_checksum = free_set_trailer.block_checksums[index],
+                        .block_address = free_set_trailer.block_addresses[index],
+                        .block_type = .free_set,
+                    };
+                } else {
+                    tour.* = .{ .client_sessions = .{} };
+                }
+            }
+
+            if (tour.* == .client_sessions) {
+                const client_sessions = scrubber.client_sessions_checkpoint;
+                if (client_sessions.callback != .none) return null;
+                if (tour.client_sessions.index < client_sessions.block_count()) {
+                    const index = tour.client_sessions.index;
+                    tour.client_sessions.index += 1;
+                    return .{
+                        .block_checksum = client_sessions.block_checksums[index],
+                        .block_address = client_sessions.block_addresses[index],
+                        .block_type = .client_sessions,
+                    };
+                } else {
+                    tour.* = .done;
+                }
+            }
+
+            // Wrap around to the next cycle.
+            assert(tour.* == .done);
+            tour.* = .init;
```

Comment:
maybe `log.info` here, so the operator can double-check that the scrubber scrubs.